QUARKUS-2812.md

```diff
@@ -0,0 +1,85 @@
+# QUARKUS-2812 Build time metrics for Quarkus
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-2812.
+
+The aim of this feature is to collect build-time data in a Quarkus application and send them to an external data platform
+(segment.io) for analytical purposes.
+The application collects various data (see bellow) during the build, logs them and sends them to the 3rd party service.
+It also logs the time the entire metrics-related operation takes.
+The functionality can be deactivated by:
+- Quarkus configuration property.
+- Static list of application artifact group ID patterns (`io.quarkus*`, `io.quarkiverse`, `org.acme`, `org.test`).
+- Dynamic remote config.
+  - Retrieved from a remote endpoint, locally cached (`{user.home}/.redhat/`). The local cache has a refresh interval.
+  - Config properties, which may be used to disable the metrics:
+    - All on/off.
+    - Deny for a list of user IDs.
+    - Deny for a list of Quarkus versions.
+
+Build failure prevents the metrics to be processed.
+Failure in the metrics collection (i.e. a bug in Quarkus) causes the build to fail.
+Failure in the metrics sendoff (network issue, endpoint unavailability, ...) does NOT cause the build to fail.
+
+## Scope of testing
+
+What data is collected in what event.
+- In all events, identical set of data is being collected.
+- Events:
+  - Dev mode build.
```

Comment:
What about tests with `@QuarkusTest`?

Reply:
`@QuarkusTest` gets executed in the `test` phase, while the metrics are being triggered by `quarkus-maven-plugin` in:
- `prepare-package` phase for dev mode (`DevMojo`)
- `package` phase for build (`BuildMojo`)

So no metrics during the tests.